packages/runtime/src/runtime.ts

```diff
@@ -802,22 +803,40 @@ export class Runtime {
    * @param debugStack: if passed then TEAL Stack is logged to console after
    * each opcode execution (upto depth = debugStack)
    */
-  executeTx (txnParams: types.ExecParams | types.ExecParams[], debugStack?: number): TxReceipt | TxReceipt[] {
-    const txnParameters = Array.isArray(txnParams) ? txnParams : [txnParams];
-    for (const txn of txnParameters) {
-      switch (txn.type) {
-        case types.TransactionType.DeployASA: {
-          if (txn.asaDef === undefined) txn.asaDef = this.loadedAssetsDefs[txn.asaName];
-          break;
-        }
-        case types.TransactionType.DeployApp: {
-          txn.approvalProg = new Uint8Array(32); // mock approval program
-          txn.clearProg = new Uint8Array(32); // mock clear program
-          break;
+  executeTx (
+    txnParams: types.ExecParams | types.ExecParams[] | types.TransactionAndSign | types.TransactionAndSign[],
```

Comment:
we should create a type for this, and use in both `algob.executeTransaction` & `runtime.executeTx`

Reply:
Can we do it on [this task](https://www.pivotaltracker.com/n/projects/2452320/stories/181295625)? I think it more better.

Reply:
it's a one liner. but OK, as you prefer